core/trino-main/src/test/java/io/trino/block/BlockAssertions.java

```diff
@@ -129,15 +127,15 @@ public static DictionaryBlock createRandomDictionaryBlock(Block dictionary, int
         checkArgument(dictionary.getPositionCount() > 0, "dictionary position count %s is less than or equal to 0", dictionary.getPositionCount());
 
         int[] ids = IntStream.range(0, positionCount)
-                .map(i -> RANDOM.nextInt(dictionary.getPositionCount()))
+                .map(i -> random().nextInt(dictionary.getPositionCount()))
```

Comment:
Um, this actually uses a fresh `Random` instance per iteration. Each instance has the same seed, so each iteration will produce the same result. So this is kind of more deterministic than you wanted, I think :)

Reply:
:facepalm: https://github.com/trinodb/trino/pull/13029 cc @sopel39